application/models/meetingparticipant.go

```diff
@@ -72,3 +83,54 @@ func (m *MeetingParticipant) CreateFromInvite(invite MeetingInvite, userID int)
 func (m *MeetingParticipant) Destroy() error {
 	return DB.Destroy(m)
 }
+
+// Create a new MeetingParticipant from a meeting ID and code. If `code` is nil, the meeting must be non-INVITE_ONLY.
+// Otherwise, `code` must match either a MeetingInvite secret code or a Meeting invite code.
+func (m *MeetingParticipant) Create(ctx context.Context, meeting Meeting, code *string) error {
```

Comment:
Should this be named `FindOrCreate` because of line 97?

Reply:
The goal was to avoid an error if the participant already exists. Maybe this means the method is trying to do too much. I think I'll pull out that "find" code and let the caller do it. Sound good?

Reply:
On second thought, I like your suggestion better. I can't think of any reason to have a `Create` method that fails when it tries to add a duplicate.